deprecatedflags/deprecatedflags.go

```diff
@@ -29,28 +30,29 @@ import (
 // and instead using the option triggers a deprecation warning. The
 // return code can be safely ignored and is only provided to support
 // replacing functions like flag.Int in a global variable section.
-func Add(name string) bool {
-	flag.Var(deprecated{name: name}, name, "This option is deprecated.")
+func Add(logger klog.Logger, name string) bool {
```

Comment:
What logger would a caller pass in here? Logging hasn't been initialized yet, so it could only pass `klog.Background`, which is already what the code below uses via `klog.Warningf`.

Logging during command line passing never works well because of that. I suggest to replace `klog.Warningf` with `fmt.Fprintf(os.Stderr, ...`

Reply:
changed: https://github.com/kubernetes-csi/csi-lib-utils/pull/149/commits/d0c5c66abd0ac95a8f5d62e2b36a3beaa3cc5f59#diff-8a66eb22e577246eefa65671fa6c956c8e869e15f6961a8faf8d32ebbd168110L55-R53